### DISCUSSION TITLE
How added to dropdown list st-link

### DISCUSSION THREAD

**Discussion by brightproject** [stm32duino/Arduino_Core_STM32#1967] [Q&A] [answered]
How added to dropdown list string ST-LINK-V2?
I have an old downloaded arduino core to work with STM32 and in it I select the firmware via st-link-v2.
![old_core_arduino](https://user-images.githubusercontent.com/1788098/223716890-d397be1d-f298-4c9b-a677-16a4e4a1b2b9.png)
But after loading a new kernel from ST, in the menu
![core_new](https://user-images.githubusercontent.com/1788098/223717929-cacd09c4-78ae-492f-8c93-03d6a96c3abd.png)
STM32CubeProgrammer(SWD)  - doesn't work either.
The compilation of the same blink takes a minute, which is not normal.
Kernel 2.4.0 installed on the recommendation through the board manager arduino IDE
https://docs.arduino.cc/learn/starting-guide/cores
In general, I can’t even choose the right board from the list, although I have a standard "blue pill".
Please point me in the right direction to figure it out.
While I'm working on the old mapple core.
![old_new](https://user-images.githubusercontent.com/1788098/223719982-36f1cda3-2281-4356-8b1d-114f85145ad1.png)
The green arrow is the old and working kernel.
Red arrow - installed and not working.

**Comment by fpistm** [comment]
> How added to dropdown list string ST-LINK-V2? I have an old downloaded arduino core to work with STM32 and in it I select the firmware via st-link-v2.

Those core are fully different. If you use STLink then select STM32CubeProgrammer (SWD) but you have to install it as described here:
https://github.com/stm32duino/Arduino_Core_STM32/wiki/Upload-methods#requirement



> The compilation of the same blink takes a minute, which is not normal.

You can't compare both core. This one provides support of 'all' stm32 series. And on windows it is a know "issue" mainly linked to IDE and Host OS windows (antivirus, network access,...) , on Linux same build goes faster and we could not do anything for this.

> Kernel 2.4.0 installed on the recommendation through the board manager arduino IDE https://docs.arduino.cc/learn/starting-guide/cores In general, I can’t even choose the right board from the list, although I have a standard "blue pill". Please point me in the right direction to figure it out. 

Select GenericF1xx then in the board part number you will find the BluePill.
![image](https://user-images.githubusercontent.com/20641798/223725192-102f549c-ab66-4761-a582-c2e713f2d7cb.png)

**Reply by brightproject** [reply]
> on windows it is a know "issue" mainly linked to IDE and Host OS windows (antivirus, network access,...)

How can the compilation speed be increased?
Use command line?
Your core seems to have its own CLI by CMake?
There is unforgivably little information about it.
Documentation only, no examples.
On YouTube and on the Internet there are no details at all.
Is this tool worth looking into or is it just "as is"?

**Reply by fpistm** [reply]
> How can the compilation speed be increased? 
Use Linux 😜 

> Use command line? Your core seems to have its own CLI by CMake? There is unforgivably little information about it. Documentation only, no examples. 

https://github.com/stm32duino/Arduino_Core_STM32/wiki/CMake-presentation

As stated: 

> Didactic examples can be found on a separate repository: https://github.com/stm32duino/CMake_workspace.

Note that it is for advanced user.

**Comment by brightproject** [comment]
I don't really like stm cube programmer
I managed to add my st-link-v2 to the menu.
And it works!
![st-link-ok](https://user-images.githubusercontent.com/1788098/223758830-0f57dee1-d787-43c5-bf91-3097b7442773.png)

But I have difference between types of programming
![STLink-V2](https://user-images.githubusercontent.com/1788098/223761422-15e8da1e-2c86-45e2-8d34-3a814ac4affb.png)
![STM32CubeProgrammer (SWD)](https://user-images.githubusercontent.com/1788098/223761429-f2847fb0-75af-4f32-a8fd-26af3943f898.png)

**Reply by fpistm** [reply, marked as answer]
> I don't really like stm cube programmer I managed to add my st-link-v2 to the menu. And it works! 

Not surprised, you can add all what you want while it is properly declared. 😉
